flux-refineck/src/constraint_gen.rs

```diff
@@ -236,13 +237,22 @@ impl<'a, 'tcx> ConstrGen<'a, 'tcx> {
             }
         }
 
+        // check (non-closure) obligations -- the closure ones are handled in `checker` since
+        // as we have to recursively walk over their def_id bodies.
+        for pred in &obligs {
+            if let rty::ClauseKind::Projection(projection_pred) = pred.kind().skip_binder() {
+                let impl_elem =
+                    normalize_alias_ty(infcx.genv, callsite_def_id, &projection_pred.alias_ty)?;
+                infcx.subtyping(rcx, &impl_elem, &projection_pred.term);
```

Comment:
I believe this is unsound in general. We should do subtyping both ways

```suggestion
                infcx.subtyping(rcx, &impl_elem, &projection_pred.term);
                infcx.subtyping(rcx, &projection_pred.term, &impl_elem);
```